database-plugins/src/main/java/co/cask/hydrator/plugin/DBUtils.java

```diff
@@ -58,6 +61,50 @@ public static void cleanup(Class<? extends Driver> driverClass) {
   }
 
   /**
+   * Ensures that the JDBC Driver specified in configuration is available and can be loaded. Also registers it with
+   * {@link DriverManager} if it is not already registered.
+   * @return {@link JDBCDriverShim} returns null if we are able to locate an existing driver for the
+   * connection string using the DriverManager.
+   */
+  @Nullable public static JDBCDriverShim ensureJDBCDriverIsAvailable(Class<? extends Driver> jdbcDriverClass,
+                                             String connectionString,
+                                             String jdbcPluginType, String jdbcPluginName)
+    throws IllegalAccessException, InstantiationException, SQLException {
+    try {
+      DriverManager.getDriver(connectionString);
+    } catch (SQLException e) {
+      // Driver not found. We will try to register it with the DriverManager.
+      LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
+                jdbcPluginType, jdbcPluginName, jdbcDriverClass.getName(),
+                JDBCDriverShim.class.getName());
+      JDBCDriverShim driverShim = new JDBCDriverShim(jdbcDriverClass.newInstance());
+      try {
+        DBUtils.deregisterAllDrivers(jdbcDriverClass);
+      } catch (NoSuchFieldException | ClassNotFoundException e1) {
+        LOG.error("Unable to deregister JDBC Driver class {}", jdbcDriverClass);
+      }
+      DriverManager.registerDriver(driverShim);
+      return driverShim;
+    }
+    return null;
+  }
+
+  /**
+   * deregister driver shim if its not null
+   * @param driverShim
+   */
+  public static void deregisterShim(@Nullable JDBCDriverShim driverShim) {
+    try {
+      if (driverShim != null) {
+        DriverManager.deregisterDriver(driverShim);
+      }
+    } catch (SQLException e) {
+      LOG.warn("Error while de-registering JDBC drivers in ETLDBOutputFormat.", e);
```

Comment:
this warning is inaccurate.  method should throw an exception and caller should handle it rather than logging, then propagating as runtime exception.